src/edgepi/eeprom/edgepi_eeprom.py

```diff
@@ -158,46 +159,43 @@ def __read_edgepi_reserved_memory(self):
             time.sleep(0.01)
         return bytes(buff[2:buff_and_len])
 
-    def get_message_of_interest(self, msg: MessageFieldNumber = None):
+    def get_edgepi_data(self):
         """
-        This function filters out the message according to the specified field number passed as
-        parameter.
+        Read Edgepi reserved memory space and populate dataclass
         Args:
-            msg (MessageFieldNumber): protocol buffer message field index number for ListFields()
-            function
+            N/A
         Return:
-            pb message specified by the message field number. ex) if message field of DAC is passed,
-            the dac message will be returned
+            eeprom_data (EepromDataClass): dataclass containing eeprom values
         """
-        self.eeprom_layout.ParseFromString(self.__read_edgepi_reserved_memory())
-        return self.eeprom_layout.ListFields()[msg.value - 1][1]
+        # pylint: disable=no-member
+        self.eeprom_pb.ParseFromString(self.__read_edgepi_reserved_memory())
+        eeprom_data = EepromDataClass.extract_eeprom_data(self.eeprom_pb)
+        return eeprom_data
 
-    def get_edgepi_reserved_data(self):
+    def set_edgepi_dataclass(self, module_name: EepromModuleNames, module_value):
```

Comment:
the original idea behind this method was to read, modify and write. to make it easy for the user to change a single or few dataclass attributes. but the current implementation is just reading and modifying which is not the intended behavior.

also I don't think it is a good idea to use a signature like `module_name: EepromModuleNames, module_value`.
include all the possible attributes like `serial_number: str = None, model: str: None, ...` and set the attributes that are not None. 

or we can just remove this method and we clearly mention in docs that you need to read the data, modify the object and write it back. I think we should do that. and also let's rename `set_edgepi_data` to `write_edgepi_data` and `get_edgepi_data` to `read_edgepi_data` to be consistent and be more accurate.

Reply:
I think the suggested method of changing few attributes will complicate the implementation due to the nested EepromDataClass. Ex) DAC calibs and ADC calibs.

I'd feel like reading dataclass then writing would be easier for the user since the LSP will help them to show which attributes are available inside the dataclass.

Reply:
I removed removed set_edgepi_dataclass and renamed get/set to read/write.

Readme is updated with a Note